### PR TITLE
gh #115 dsAudio: L3 Test case Development

### PR DESCRIPTION
**Pull request by bhanucbp**
Added dsAudio L3 Testcases

### CODE REVIEW

File: src/test_l3_dsAudio.c [hunk 1]
```diff
@@ -0,0 +1,3045 @@
...
+  { "true",  (int32_t)true  },
+  {  NULL, -1 }
+};
+
+/**
+ * @brief This function clears the stdin buffer.
+ *
+ * This function clears the stdin buffer.
+ */
+void readAndDiscardRestOfLine(FILE* in)
```
Comment (Ulrond) [inline src/test_l3_dsAudio.c:199]: can you raise a ticket on ut-core to support this.

File: src/test_l3_dsAudio.c [hunk 2]
```diff
@@ -0,0 +1,3045 @@
...
+}
+
+/**
+ * @brief Callback function for Headphone connection status.
+ *
+ * This function is invoked whenever a change occurs in the Headphone connection.
+ */
+static void audioOutPortConnectCB(dsAudioPortType_t portType, uint32_t uiPortNo, bool isPortCon)
+{
+    UT_LOG_INFO("Received Connection status callback port: %s, port number: %d, Connection: %s",
```
Comment (Ulrond) [inline src/test_l3_dsAudio.c:212]: we can try printing this to the terminal, but I think the the future you may want to specify via the menu an optional log filename that you write too.

don't need to change it now, but something for consideration

File: src/test_l3_dsAudio.c [hunk 3]
```diff
@@ -0,0 +1,3045 @@
...
+}
+
+/**
+ * @brief Callback function for detecting audio format changes.
+ *
+ * This function is triggered whenever the audio format of the playback stream changes.
+ */
+static void audioFormatUpdateCB(dsAudioFormat_t audioFormat)
+{
+    UT_LOG_INFO("Received Format update callback : %s", UT_Control_GetMapString(dsAudioFormat_mapTable, audioFormat));
```
Comment (Ulrond) [inline src/test_l3_dsAudio.c:225]: we can try printing this to the terminal, but I think the the future you may want to specify via the menu an optional log filename that you write too.

don't need to change it now, but something for consideration

**Review by Ulrond — DISMISSED** [review]
think it's a good start to play with.

**Review by hari22yuva — APPROVED** [review]
Initial version is ok

**Pull request merged**